src/GithubMergeTool/GithubMergeTool.cs

```diff
@@ -97,13 +93,54 @@ public class GithubMergeTool
             var jsonBody = JObject.Parse(await response.Content.ReadAsStringAsync());
             if (jsonBody.Type == JTokenType.Array)
             {
-                // Branch doesn't exist
+                // Source branch doesn't exist
                 return (false, response);
             }
 
             var srcSha = ((JValue)jsonBody["object"]["sha"]).ToObject<string>();
 
-            // Create a branch on the repo
+            var existingPR = JArray.Parse(await prsResponse.Content.ReadAsStringAsync()).FirstOrDefault(pr => (string)pr["title"] == prTitle);
+            if (existingPR != null)
+            {
+                if (updateExistingPr)
+                {
+                    // Get the SHA of the PR branch HEAD
+                    var prSha = ((JValue)existingPR["head"]["sha"]).ToObject<string>();
+                    var existingPrNumber = ((JValue)existingPR["number"]).ToObject<string>();
+
+                    // Check for merge conflicts
+                    var existingPrMergeable = await IsPrMergeable(existingPrNumber);
+
+                    // Only update PR w/o merge conflicts
+                    if (existingPrMergeable == true && prSha != srcSha)
+                    {
+                        // Try to reset the HEAD of PR branch to latest source branch
+                        response = await ResetBranch(prBranchName, srcSha, force: false);
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            Console.WriteLine($"There's additional change in `{srcBranch}` but an attempt to fast-forward `{prBranchName}` failed.");
+                            return (false, response);
+                        }
+
+                        await PostComment(existingPrNumber, $"Reset HEAD of `{prBranchName}` to `{srcSha}`");
+
+                        // Check for merge conflicts again after reset.
+                        existingPrMergeable = await IsPrMergeable(existingPrNumber);
+                    }
+
+                    // Add label if there's merge conflicts even if we made no change to merge branch,
+                    // since can also be introduced by change in destination branch. It's no-op if the
+                    // label already exists. 
+                    if (existingPrMergeable == false)
```

Comment:
@RikkiGibson We are now checking mergeable state directly, and add "merge conflicts" label accordingly during update too.